extensions/amp-form/0.1/amp-form.js

```diff
@@ -64,6 +64,17 @@ export class AmpForm {
           'form action-xhr should not be on cdn.ampproject.org: %s',
           this.form_);
     }
+
+    const submitButtons = this.form_.querySelectorAll('input[type=submit]');
```

Comment:
Not sure where the efficiency and robustness win would come from?

Also, this is the same as earlier, we won't have an early error thrown when there are no submit buttons.